src/Core/Vipr/FileWriter.cs

```diff
@@ -26,8 +31,37 @@ public static void Write(IEnumerable<TextFile> textFilesToWrite, string outputDi
                 if (!Directory.Exists(Path.GetDirectoryName(filePath)))
                     Directory.CreateDirectory(Path.GetDirectoryName(filePath));
 
-                File.WriteAllText(filePath, file.Contents);
+                fileTasks.Add(WriteToDisk(filePath, file.Contents));
             }
+            await Task.WhenAll(fileTasks);
+        }
+
+        /**
+         * Write the file to disk. If the file is locked for editing,
+         * sleep until available
+         */
+        public static async Task WriteToDisk(string filePath, string output)
+        {
+            for (int tries = 0; tries < 10; tries++)
+            {
+                StreamWriter sw = null;

```

Comment:
Another note is you may want to lock based on the file path??? This means more streamwriters could be created if it is a different filepath.